pkg/avfoundation/avfoundation_darwin.go

```diff
@@ -141,6 +147,8 @@ func (rc *ReadCloser) Close() {
 		rc.onClose()
 	}
 	rc.cancelFunc()
+	rc.closeWG.Wait()
+	close(rc.dataChan)
 	unregister(rc.id)
```

Comment:
I believe current code fixes all cases, but would be better to put unregister first to reduce unnecessary call which is aborted by `rc.cancelCtx.Err()` check.

Reply:
Shouldn't we also close something to avoid the callback being called at all ? I mean, unregister will just silence the the fact we receive data right ?

Reply:
Ho ok, it's because we close the avfoundation session in the code calling this close function.